src/wp-includes/post.php

```diff
@@ -6522,6 +6531,37 @@ function wp_delete_attachment_files( $post_id, $meta, $backup_sizes, $file ) {
 	return $deleted;
 }
 
+/**
+ * Delete sources files from a directory, skipping default mime type.
+ *
+ * @since 6.0.0.
+ *
+ * @param array  $sources            The sources array to delete files from.
+ * @param string $intermediate_dir   The directory to delete files from.
+ * @param string $file               Absolute path to the attachment's file.
+ * @return bool Whether deletion succeeded.
+ */
+function _wp_delete_alternate_mime_sizes( $sources, $intermediate_dir, $file ) {
+	$deleted = true;
+	$index   = 0;
+	foreach ( $sources as $mime => $properties ) {
+		// Skip the default mime type.]
```

Comment:
wonder if we should try deleting anyway, in case the filter has been changed since the images were created? if the file was already deleted, `wp_delete_file` will fail silently.

Reply:
I think we can avoid this problem by integrating this directly into `wp_delete_attachment_files`, going over the `sources` in that function (if present). We used an extra function like this in the plugin, but I think in core it's cleaner integrating with the existing function.

Reply:
Done, makes it a little cleaner